ToDoList-Redux/ToDoList_ReduxApp.swift

```diff
@@ -9,14 +9,32 @@ import SwiftUI
 
 @main
 struct ToDoList_ReduxApp: App {
-    
-    private static let store = World.origin.store()
-    
-    @StateObject var taskViewModel = TaskList.viewModel(store: store)
-    
+    @StateObject var store = World.origin.store().asObservableViewModel(initialState: .empty)
+
     var body: some Scene {
         WindowGroup {
-            TaskList(viewModel: taskViewModel)
+            Router.taskListView(store: store)
         }
     }
 }
+
+import SwiftRex
+struct Router {
+    static func taskListView<S: StoreType>(store: S) -> TaskList
+    where S.StateType == AppState, S.ActionType == AppAction {
+        TaskList(
+            viewModel: TaskList.viewModel(store: store.projection(action: AppAction.task, state: \AppState.tasks)),
+            rowView: { id in taskListRowView(store: store, taskId: id) }
```

Comment:
This is the view producer injection. This is useful because now we have the full store available when creating any view, without having to send it to the view itself. The closure environment receives the full store, but the view itself doesn't need it. This is specially useful when using views from different modules or when using NavigationLink to push views with completely different requirements.

This also decouples more the parent and child view, as the parent doesn't have to either hold more state than necessary (we were able to have an row that only needs Task Id and nothing else) nor handle more events than necessary (we were able to remove "toggle" action from the parent, and the proxying of it).